test/pool.test.ts

```diff
@@ -1136,4 +1307,75 @@ describe("Erc20Pool", function () {
     await mineBlocksUntilCycleEnd();
     await receiver.collect(10);
   });
+
+  it("Allows full sender update with top up", async function () {
```

Comment:
One thing we might want to consider for the future is to abstract the test cases over the concrete pool implementation. Meaning that we would write the test cases once for an abstract pool and run them against both the ERC20 and the Eth pool.

Reply:
Brilliant idea worth exploring!